File: spring-integration/subscribable-channels/messaging-bridge/src/main/java/com/ddubson/integration/services/UppercasePrintService.java
package com.ddubson.integration.services;

import com.ddubson.ANSIColor;
import com.ddubson.LogAdapter;
import org.springframework.messaging.Message;

import static java.lang.String.format;

public class UppercasePrintService {
    private final LogAdapter logAdapter;

    public UppercasePrintService(LogAdapter logAdapter) {
        this.logAdapter = logAdapter;
    }

    public void print(Message<String> msg) {
        logAdapter.info(
                format("[2] Consuming pub-sub channel -- [%s] %s", Thread.currentThread().getName(),
                        msg.getPayload().toUpperCase()), ANSIColor.ANSI_CYAN);
    }
}

File: spring-integration/subscribable-channels/direct-channel/src/main/kotlin/com/ddubson/integration/services/UppercasePrintService.kt
package com.ddubson.integration.services;

import com.ddubson.ANSIColor
import com.ddubson.LogAdapter
import org.springframework.messaging.Message

class UppercasePrintService(private val logAdapter: LogAdapter) {
    fun print(msg: Message<String>) {
        logAdapter
                .info("[2] Consuming input channel -- [${Thread.currentThread().name}] ${msg.payload.toUpperCase()}",
                        ANSIColor.ANSI_PURPLE)
    }
}
